### PR TITLE
hide-flyout: new setting "open by default" (auto-hide block pallete addon)

### PR DESCRIPTION
**Pull request by callumjt**
adds a new option to Auto-hiding block palette which makes the flyout lock when the page is loaded 

tested on chrome and firefox :)

(sorry for low framerate, for some reason obs decided to die)

https://github.com/ScratchAddons/ScratchAddons/assets/116295240/e93ebaf6-017d-46e5-8193-a5f20e165a07

### CODE REVIEW

**Comment by CubesterYT** [comment]
I remember this was asked for in TurboWarp, I'm so stoked this is coming, I need it bad.

**Comment by LilyMakesThings** [comment]
A dream come true, hope to see this in TurboWarp ASAP <3

**Review by DNin01 — CHANGES_REQUESTED** [review]
Pretty cool, but this does need testing with the "Category click" setting. There's no lock button when you use that setting, but it could instead be programmed to start out open, requiring a category click to close. Related: #5821. What actually happens in this PR is **you can't close it**.

I like to set this addon to toggle on category click and I've always wanted it to start out open.

**Comment by callumjt** [comment]
> Pretty cool, but this does need testing with the "Category click" setting. There's no lock button when you use that setting, but it could instead be programmed to start out open, requiring a category click to close. Related: #5821. What actually happens in this PR is **you can't close it**.
> 
> I like to set this addon to toggle on category click and I've always wanted it to start out open.

was making this and tried to set "motion" to enabled on game start but "Blockly.getMainWorkspace().getToolbox().selectedItem_.setSelected(false);" doesnt seem to work

**Comment by DNin01** [comment]
@mxmou, do you want to take a look?

**Comment by callumjt** [comment]
the code i used is 

`const option = addon.settings.get("lockLoad");
    if (option) {
      if (getToggleSetting() === "category") {
        Blockly.getMainWorkspace().getToolbox().selectedItem_.setSelected(true);
        onmouseleave(null, 0);
        toggle = true;
      } else {
        flyoutLock = option;
        updateLockDisplay()
      }
      flyOut.classList.remove("sa-flyoutClose");
      scrollBar.classList.remove("sa-flyoutClose");
    };`

**Comment by mxmou** [comment]
> was making this and tried to set "motion" to enabled on page load but "Blockly.getMainWorkspace().getToolbox().selectedItem_.setSelected(true);" doesnt seem to work

That's because of the `setSelected(false)` call on line 191. Try adding `&& !toggle` to the `if (!shouldScroll)` condition. You can then remove line 53.

**Review by DNin01 — APPROVED** [review]
Seems to be working!

Only thing is the block category [doesn't initially appear highlighted](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/6b86a6a9-0c81-4dc1-81e9-e82adb1c2eb1) if you've scrolled the block palette to the very top and then dynamically enable the addon using the "Category click" and "Lock by default" settings, but that's very minor.

File: addons/hide-flyout/addon.json [hunk 1]
```diff
@@ -108,6 +108,12 @@
         }
       ],
       "default": "default"
+    },
+    {
+      "name": "Lock by default",
```
Comment (Samq64) [inline addons/hide-flyout/addon.json:113]: The category click setting doesn't have a lock.
```suggestion
      "name": "Open by default",
```

**Review by Samq64 — COMMENTED** [review]
Could `latestUpdate` be added to highlight the new setting?

**Comment by callumjt** [comment]
Changed the option name and added latestUpdate

File: addons/hide-flyout/addon.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
   ],
   "dynamicEnable": true,
   "dynamicDisable": true,
+  "latestUpdate": "1.36.0",
```
Comment (Samq64) [inline addons/hide-flyout/addon.json:22]:
```suggestion
  "latestUpdate": {
    "version": "1.36.0",
    "newSettings": ["lockLoad"]
  }.
```

**Review by Samq64 — APPROVED** [review]

File: addons/hide-flyout/userscript.js [hunk 3]
```diff
@@ -46,6 +46,21 @@ export default async function ({ addon, console, msg }) {
     lockIcon.src = addon.self.dir + `/${flyoutLock ? "" : "un"}lock.svg`;
   }
 
+  function autoLock() {
+    const option = addon.settings.get("lockLoad");
+    if (option) {
+      if (getToggleSetting() === "category") {
+        onmouseleave(null, 0);
```
Comment (mxmou) [inline addons/hide-flyout/userscript.js:53]:
```suggestion
```
`onmouseleave()` closes the palette. There's no reason to call it here.

**Review by mxmou — CHANGES_REQUESTED** [review]
Lines 151-155 (in the `"reenabled"` event listener) close the palette if the addon is dynamically enabled and the mode is category click. This should only happen if the new setting is disabled.

When switching from category click to a different mode, the palette should be automatically locked if the new setting is enabled.

**Review by mxmou — APPROVED** [review]

**Pull request merged**